internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -153,14 +164,20 @@ func isEssential(name string, s convertSidecarOpts) bool {
 	return false
 }
 
-func isEssentialStatus(s string) bool {
-	for _, allowed := range essentialContainerStatus {
-		if s == allowed {
-			return true
+func isEssentialStatus(status string, container string, c convertSidecarOpts) (bool, error) {
+	if _, ok := c.sidecarConfig[container]; ok {
```

Comment:
since this bit of code is used pretty often, we might want to refactor it into a function:

```go
func isSidecar(container string, sidecarConfig map[string]*manifest.SidecarConfig) bool
```

and this line can be replaced by:
 ```suggestion
	if isSidecar(container, c.sidecarConfig) {
```

Reply:
Better modularized.